doc: Update Thread 1.2 supportted features with DUA overview. (#5458)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_supported_features.rst

Enhanced-ACK probing is the most power efficient method to retrieve link metrics since very little message overhead is required.
|NCS| provides full Link Metrics support even for the RCP architecture, which is the most technically challenging one since the radio driver must handle the injection of Information Elements on time to match the acknowledgment timing requirements.
The decision on how to interpret the link metrics information to adjust the transmission power is left to the application itself.

Thread Domain Unicast Addressing
================================

Thread 1.2 Specification introduces concept of Thread Domains.

> A Thread Domain is a set of Thread Devices that receive and apply common Thread Domain operational configuration. The Thread Domain operational configuration enables
> Thread Devices to join and participate in larger interconnected systems extending beyond the limits of a single Thread Network. A user or network administrator may use functions
> of Thread Commissioning or of Thread Border Routers to set up a common Thread Domain operational configuration for Thread Devices belonging to different Thread
> Networks or Partitions having potentially different per-network credentials.

Reviewer: ```suggestion
A Thread Domain is a set of Thread Devices that receive and apply a common Thread Domain operational configuration.
The Thread Domain operational configuration enables Thread Devices to join and participate in larger interconnected systems extending beyond the limits of a single Thread Network. 
A user or network administrator may use functions of either Thread Commissioning or Thread Border Routers to set up a common Thread Domain operational configuration for Thread Devices.
The Thread Devices can belong to different Thread Networks or Partitions that have potentially different per-network credentials.
```
I am not sure, but I feel the word "common" could be replaced here with "shared" in both instances.

Also, the text isn't clear, but does a Thread Domain have to extend beyond a single Thread Network?

Enhanced-ACK probing is the most power efficient method to retrieve link metrics since very little message overhead is required.
|NCS| provides full Link Metrics support even for the RCP architecture, which is the most technically challenging one since the radio driver must handle the injection of Information Elements on time to match the acknowledgment timing requirements.
The decision on how to interpret the link metrics information to adjust the transmission power is left to the application itself.

> Thread Domain Unicast Addressing

Reviewer 2: ```suggestion
Thread Domain unicast addressing
```

================================

Thread 1.2 Specification introduces the concept of Thread Domains.

A Thread Domain is a set of Thread Devices that receive and apply a common Thread Domain operational configuration.

> The Thread Domain operational configuration enables Thread Devices to join and participate in larger interconnected systems extending beyond the limits of a single Thread Network.

Reviewer 2: Thread Network or Thread network?
Normally I would say Thread network.
Unless there is a reason for doing otherwise.
I see we are using both ways of writing in this section.
We should align.

Reviewer 3: Let's go with _Thread network_.

> A user or network administrator may use functions of either Thread Commissioning or of Thread Border Routers to set up a common Thread Domain operational configuration for Thread Devices.

Reviewer 2: ```suggestion
A user or network administrator may use functions of either Thread Commissioning or Thread Border Routers to set up a common Thread Domain operational configuration for Thread Devices.
```

The Thread Devices can belong to different Thread Networks or Partitions that have potentially different per-network credentials.

Multicast across Thread Networks
================================

Thread 1.1 border routers have a limitation not to forward multicast traffic with scope greater than realm-local.

> For certain applications it could be useful to be able to control multicast groups from a host outside the Thread network.

Reviewer 2: Thread Network or network?

This is achieved in Thread 1.2 by allowing Thread border routers to forward multicast traffic with scope greater than realm-local in two ways:

* From the Thread network to the exterior network: as a configuration option in the border router, for every multicast group.

> * From the exterior network to the Thread network: the Primary Backbone Router would forward only multicast traffic with a destination matching one of the multicast groups registered by Thread devices in its network.

Reviewer 2: ```suggestion
* From the exterior network to the Thread network: the Primary Backbone Router (PBBR) would forward only multicast traffic with a destination matching one of the multicast groups registered by Thread devices in its network.
```

For the second case, a Thread Commissioner can be used as well to register allowed multicast groups on behalf of the devices.

> The OpenThread stack will handle automatically the registration of the multicast groups with the proper PBBR whenever they are configured in the device.

Reviewer 2: ```suggestion
The OpenThread stack will automatically handle the registration of multicast groups with proper PBBR whenever they are configured in the device.
```
